build_tools/azure/install.sh

```diff
@@ -126,17 +126,23 @@ scikit_learn_install() {
         export LDFLAGS="$LDFLAGS -Wl,--sysroot=/"
     fi
 
-    if [[ "$BUILD_WITH_MESON" == "true" ]]; then
-        make dev-meson
+    if [[ "$BUILD_WITH_SETUPTOOLS" == "true" ]]; then
+        python setup.py develop
     elif [[ "$PIP_BUILD_ISOLATION" == "true" ]]; then
         # Check that pip can automatically build scikit-learn with the build
         # dependencies specified in pyproject.toml using an isolated build
         # environment:
-        pip install --verbose --editable .
+        pip install --verbose .
     else
+        if [[ "$UNAMESTR" == "MINGW64"* ]]; then
+           # Needed on Windows CI to compile with Visual Studio compiler
+           # otherwise Meson detects a MINGW64 platform and use MINGW64
+           # toolchain
+           ADDITIONAL_PIP_OPTIONS='-Csetup-args=--vsenv'
```

Comment:
This is duplicated several times. Isn't there a way to specify that in a meson config file or in `pyproject.toml`?

Not a big deal if it's not possible with a short config change.

Reply:
So it looks like this is possible in `pyproject.toml` https://meson-python.readthedocs.io/en/stable/how-to-guides/meson-args.html#force-the-use-of-the-msvc-compilers-on-windows

Numpy does not seem to use it.

I am wondering whether there is some use case where you may want to compile with MINGW although I am not an expert on this ...

Reply:
Yes, there's people building with MinGW (e.g. for Cygwin) and also folks using Intel compilers. Forcing MSVC even when other compilers are on the path may have downsides. The detailed behavior is described at https://mesonbuild.com/Builtin-options.html#details-for-vsenv. The main one I think is _"It also make Meson abort with an error message when activation fails."_ On GitHub Actions that's always installed, but locally it may not be. So that would then force the user to patch out the hardcoded `--vsenv`.

There won't be too many users who build from source on Windows but don't have MSVC installed, so it may still be a valid choice. The upside is less code duplication (you don't have to specify `--vsenv` elsewhere) and that some other users who do have MinGW/Intel compilers on the path may still expect MSVC to be selected first (IIRC this is what `distutils` does).

Reply:
Thanks a lot for the details! I would be inclined to keep the `--vsenv` in multiple places as we (and numpy as well) currently do.